src/main/java/com/ryszardpanda/Library/repository/BookRepository.java

```diff
@@ -0,0 +1,95 @@
+package com.ryszardpanda.Library.repository;
+
+import com.ryszardpanda.Library.model.BookStats;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+import com.ryszardpanda.Library.model.Book;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Repository
+@RequiredArgsConstructor
+public class BookRepository {
+
+    private final List<Book> books;
+
+    public Book addBook(Book book) {
+        book.setId(books.size() + 1);
+        books.add(book);
+        return book;
+    }
+
+    public List<Book> getAllBooks() {
+        return new ArrayList<>(books);
+    }
+
+    public Optional<Book> getBookById(Integer id) {
+        if (id == null) {
+            return Optional.empty();
+        }
+        return books.stream()
+                .filter(book -> book.getId().equals(id))
+                .findFirst();
+    }
+
+    public Optional<Book> updateBookById(Integer id, Book updatedBook) {
+        Optional<Book> optionalBook = getBookById(id);
+        optionalBook.ifPresent(book -> {
+            book.setBookTitle(updatedBook.getBookTitle());
+            book.setBookAuthor(updatedBook.getBookAuthor());
+            book.setPublishYear(updatedBook.getPublishYear());
+            book.setRating(updatedBook.getRating());
+        });
+        return optionalBook;
+    }
+
+    public boolean deleteBookById(Integer id) {
+        return books.removeIf(book -> book.getId().equals(id));
+    }
+
+    public List<Book> searchBooksByTitle(String title) {
+        return books.stream()
+                .filter(book -> book.getBookTitle().contains(title))
+                .collect(Collectors.toList());
+    }
+
+    public List<Book> searchBooksByAuthorAndYear(String author, Integer year) {
+        return books.stream()
```

Comment:
Wyobrazmy sobie ze mamy 3 ksiazki o author == null i year == null 
to wtedy 
te ksaizki wszystkie przejda i zostan zwrocone bo:
.filter(book -> (author == null || book.getBookAuthor().contains(author)))
                .filter(book -> (year == null || book.getPublishYear().equals(year)))
     
                 .filter(book -> (author != null && book.getBookAuthor().contains(author)) || 
                 (year != null && book.getPublishYear().equals(year)) ) <- zastanow sie nad tym i potestuj